### PR TITLE
DEV2-2924 fix delayed capabilities issue

### PR DESCRIPTION
**Pull request by avnerbarr**

### CODE REVIEW

File: src/registration/popupTristate.ts [hunk 1]
```diff
@@ -32,28 +29,21 @@ export class PopupTristate implements Disposable {
...
       });
     }
   }
 
   get needWait(): boolean {
-    return (
-      this.installationState === InstallationState.Undefined ||
-      !statePoller.state.currentState
-    );
+    return this.newInstall === undefined || this.isLoggedIn === undefined;
```
Comment (yanir-codota) [inline src/registration/popupTristate.ts:42]: What is the effect of this? I'm missing context here.

File: src/registration/forceRegistration.ts [hunk 2]
```diff
@@ -35,20 +35,21 @@ export function shouldStatusBarBeProminent(): boolean {
 
 if (isForceEnabled.value === undefined) {
   const disposable = onDidRefreshCapabilities(() => {
-    disposable.dispose();
     isForceEnabled.value = isCapabilityEnabled(Capability.FORCE_REGISTRATION);
+    if (isForceEnabled.value) {
+      disposable.dispose();
```
Comment (yanir-codota) [inline src/registration/forceRegistration.ts:40]: What is this disposable object?

**Review by yanir-codota — COMMENTED** [review]
I need more context. 
I feel I'm missing some of the lifecycle of this plug-in.

File: src/registration/popupTristate.ts (hunk 1 above)
Comment (avnerbarr) [reply src/registration/popupTristate.ts:42]: I suggest you read the code :)

File: src/registration/forceRegistration.ts (hunk 2 above)
Comment (avnerbarr) [reply src/registration/forceRegistration.ts:40]: this is the design pattern implemnted in vscode

File: src/registration/popupTristate.ts (hunk 1 above)
Comment (avnerbarr) [reply src/registration/popupTristate.ts:42]: the capabilities arrive late

the state is determined by events that occur asynchronously. This code waits for the collection of events which is to trigger the relevant notifications

Comment (yanir-codota) [reply src/registration/popupTristate.ts:42]: 👍

**Review by yanir-codota — APPROVED** [review]

**Pull request merged**